test/e2e/providers_test.go

```diff
@@ -0,0 +1,120 @@
+package e2e
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	operatorsv1 "github.com/openshift/api/operator/v1"
+	consoleapi "github.com/openshift/console-operator/pkg/api"
+	"github.com/openshift/console-operator/pkg/testframework"
+)
+
+const (
+	statuspageIDField = "statuspageID"
+	providersField    = "providers"
+)
+
+func setupProvidersTestCase(t *testing.T) *testframework.Clientset {
+	client := testframework.MustNewClientset(t, nil)
```

Comment:
Mostly I'm mentioning this to ensure a `tearDown()` func puts the `config` resources back in a `vanilla` state, and that it `poll` the `operator` `config` for `Available:True, Progressing:False, Failing:False` to ensure we are ready to go on to the next test (avoid flakes).

@zherman0 was working on a `testframework` generic `poll`.

Reply:
I don't think we have to hold this PR for it, but we will have to go back and update them when the func is ready.